### DISCUSSION TITLE
How to align nodes horizontally (sequence way)

### DISCUSSION THREAD

**Discussion by susmithakundukulam** [d3/d3#3894] [Q&A] [answered]
Hi,

I am new to D3 library. Before moving further just to know if is it possible to generate below diagram. 
![image](https://github.com/d3/d3-hierarchy/assets/31012506/0b12cd71-b564-4f82-a0b2-9ae7fe0be8e0)

Just note like these exact double lines are mandatory . Should be able to connect sequentially like B,A,C,D[same horizontal way]. And also can have subnodes from each like F,H,I and sequence diagram again follows from these nodes  which is optional (without disturbing other connecting lines).

Is this possible ? Thanks in advance

**Comment by Fil** [comment, marked as answer]
Yes, D3 leaves you free to draw the lines however you want.

[d3-hierarchy](https://d3js.org/d3-hierarchy/tree) offers algorithms that can lay out a tree (deciding where each node goes) according to certain rules. But it is up to you to decide if you want to draw them at the resulting positions, or if you want to draw lines for each connector "parent to child", or other lines.

If you pass a hierarchy where 

* B, A, C, D  are children of root
* E, F, G are children of C ; H, I, J children of D

then the nodes will be placed where you want them.

Then you can draw lines between them representing other kinds of relations.

If your requirements are more complex, as an alternative we recommend [GraphViz](https://www.graphviz.org/gallery/), which has many (very many) layout algorithms.

**Reply by susmithakundukulam** [reply]
Hi,

Thank your for the repsonse

Even if I pass the hierarchy as mentioned. how will we manage to draw those double lines? Here those double connections are mentioned as partners. Can we manage the generations as well.?..
